api-sandbox/Web3.elm

```diff
@@ -40,3 +50,32 @@ type BasicFunction
 
 type EthFunction
     = GetBlockNumber
+
+
+handleResponse : Web3State msg -> Int -> Maybe (String -> msg)
+handleResponse (Web3State state) id =
+    Dict.get id state
+
+
+getBlockNumber : Web3State msg -> (String -> msg) -> ( Web3State msg, Cmd msg )
+getBlockNumber (Web3State state) msg =
+    let
+        id =
+            1
+
+        state_ =
+            Dict.insert id msg state
+    in
+        ( Web3State state_
+        , request
+            { func = "eth.getBlockNumber"
+            , args = []
+            , id = id
+            }
+        )
+
+
+port request : Request -> Cmd msg
```

Comment:
These could probably be named a bit better. Just wanted to move them into the `Web3` module to keep `Main` simpler